lua/lualine.lua

```diff
@@ -150,32 +152,60 @@ local function statusline(sections, is_focused)
   if M.theme ~= theme_set then
     set_lualine_theme()
   end
-  local status = {}
-  if sections.lualine_a then
-    local hl = highlight.format_highlight(is_focused, 'lualine_a')
-    table.insert(status, utils_component.draw_section(sections.lualine_a,  hl))
-  end
-  if sections.lualine_b then
-    local hl = highlight.format_highlight(is_focused, 'lualine_b')
-    table.insert(status, utils_component.draw_section(sections.lualine_b,  hl))
-  end
-  if sections.lualine_c then
-    local hl = highlight.format_highlight(is_focused, 'lualine_c')
-    table.insert(status, utils_component.draw_section(sections.lualine_c,  hl))
-  end
-  table.insert(status, "%=")
-  if sections.lualine_x then
-    local hl = highlight.format_highlight(is_focused, 'lualine_c')
-    table.insert(status, utils_component.draw_section(sections.lualine_x,  hl))
-  end
-  if sections.lualine_y then
-    local hl = highlight.format_highlight(is_focused, 'lualine_b')
-    table.insert(status, utils_component.draw_section(sections.lualine_y,  hl))
+  -- status_builder stores statusline without section_separators
+  local status_builder = {}
+  -- The sequence sections should maintain
+  local section_sequence = {'a', 'b', 'c', 'x', 'y', 'z'}
+
+  for _, sec in ipairs(section_sequence) do
+    if sections['lualine_'..sec] then
+      -- insert highlight+components of this section to status_builder
+      local highlight = highlight.format_highlight(is_focused,
```

Comment:
redefined local highlight, yikes

Reply:
I was kind of running out of names lol. I'll rename it any suggestion?